numba/unicode.py

```diff
@@ -622,6 +622,29 @@ def join_str_impl(sep, parts):
         return join_str_impl
 
 
+@overload_method(types.UnicodeType, 'zfill')
+def unicode_zfill(str_to_fill, width_to_fill):
+    def zfill_impl(fstr, fwidth):
```

Comment:
I just played around with the code a bit more and I believe it would be advantageous to add some type inference here. Documentation for `overload` and friends is currently being written, so you can have a look here: https://github.com/numba/numba/pull/3951

In essence, what it comes down to is something along the lines of:

```
if not isinstance(width types.Integer):
    raise TypingError("'width' must be an Integer")
```

This will issue a better and more helpful error message for the user.

Reply:
Hint: you need to do this *before*  the implementation code.